### DISCUSSION TITLE
Do form actions encrypt what they post to the +page.server.ts?

### DISCUSSION THREAD

**Discussion by OsmanKaan58** [sveltejs/kit#11891] [Q&A] [answered]
Hi,
I am very new to sveltekit and I am curious about the default security of form actions. When a user submits a form does the data posted from the client to the server using svelte form actions get encrypted? If not how can I use RSA for example to encrypt user password which is posted from the client to my server(page.server.ts) through form actions?

**Comment by aloker** [comment, marked as answer]
Make sure your connection is TLS encrypted (is, you only ever use https) and you'll be fine. I wouldn't try to encrypt form data on the client, see eg https://stackoverflow.com/questions/39199960/should-i-encrypt-login-password-on-javascript-or-on-backend

**Reply by OsmanKaan58** [reply]
Thank you
